GliaWidgets/Sources/Download/FileDownload.Environment.Interface.swift

```diff
@@ -38,4 +39,22 @@ extension FileDownload.Environment.FetchFile {
                 return nil
             }
         }
+
+    static func fetchForEngagementFile(
+        _ file: CoreSdkClient.EngagementFile,
+        environment: Environment
+    ) -> FileDownload.Environment.FetchFile {
+        guard let components = file.url
+            .flatMap({ url in URLComponents(url: url, resolvingAgainstBaseURL: false) }),
+            components.path.starts(with: "/messaging/files/") else {
+            return .fromEngagement(environment.fetchFile)
```

Comment:
Do you think that it is advisable to have a third type, something like unknown, in case the file doesn't have `/messaging/` or `/engagement/`? Or it's not necessary, as the SDK is already kind of working like that anyways at the moment? Just wondering about any slight possibility of us downloading malicious files.

Reply:
I think it is already does quite a lot as of Widgets side. Maybe that case needs to be done on CoreSDK or server side. I'm not sure if any protection for open source code like this can do much.